usaspending_api/awards/models.py

```diff
@@ -310,6 +328,24 @@ def get_default_fields(path=None):
             "assistance_data"  # must match related_name in TransactionAssistance
         ]
 
+    @classmethod
+    def get_or_create(cls, **kwargs):
+        """Gets and updates, or creates, a Transaction
+
+        Transactions must be unique on Award, Awarding Agency, and Mod Number
+        """
+        transaction = cls.objects.filter(
+            award=kwargs.get('award'),
+            awarding_agency=kwargs.get('awarding_agency'),
```

Comment:
_In theory_, the awarding agency is part of the award. So, it might not be necessary to include `awarding_agency` in this lookup.

Pinging @kaitlin because awards and awarding agencies have been vexing lately, and would be grateful for her .02.